### DISCUSSION TITLE
send transaction using web3modal

### DISCUSSION THREAD

**Discussion by FarhaE** [WalletConnect/walletconnect-monorepo#4120] [Developer Support]
```dart
Future<String> submit(String functionName, List<dynamic> args) async {
  //EthPrivateKey credentials = EthPrivateKey.fromHex('***************');
  DeployedContract contract = await loadContract();
  final ethFunction = contract.function(functionName);
  final result = await ethClient.sendTransaction(
    credentials,
    Transaction.callContract(
      contract: contract,
      function: ethFunction,
      parameters: args,
    ),
    chainId: int.parse(w3mProvider.w3mService.selectedChain!.chainId),
  );

  return result;
}
```
How can I do this sendtransaction using web3Modal, without using private key.

**Comment by glitch-txs** [comment]
you need to use wagmi actions: https://1.x.wagmi.sh/core/actions/sendTransaction

**Comment by FarhaE** [comment]
Thanks, but my code is in Flutter, and I believe the code above is for react. Could you please provide something similar for flutter?

**Reply by glitch-txs** [reply]
@quetool could you check? 🙏

**Reply by GAEAlimited** [reply]
https://github.com/orgs/WalletConnect/discussions/4120#discussioncomment-8141561

**Comment by quetool** [comment]
Hello @FarhaE , what do you want to do exactly? send/sign a transaction or interact with a Smart Contract?

**Reply by FarhaE** [reply]
Hi, "If you want to integrate web3dart with other wallet providers, you can implement Credentials and override the appropriate methods", this is what I am trying to do. I am trying to run a smart contract function(which requires gas fee) using my web3wallet. Currently, the function runs by taking the gas fee automatically, 
So in short,  is there a way to send transactions without providing the private key and just by using the web3wallet?(Apologies if this sounds confusing)

**Reply by quetool** [reply]
Hello @FarhaE ! Oh ok I get it now. Sadly as of today the only way to interact with smart contracts is by relying on `web3dart`, which is a thirty party plug-in that uses private keys. You can refer to their docs on how to do it. (also check out [this thread](https://github.com/WalletConnect/WalletConnectFlutterV2/issues/246#issuecomment-1884650667))
We are going to start working on our own methods soon. Thanks!

**Reply by FarhaE** [reply]
Ok, Thank You!

**Reply by quetool** [reply]
Hello @FarhaE ! Could you check this reply? It might help https://github.com/WalletConnect/WalletConnectFlutterV2/issues/246#issuecomment-1900855332

**Comment by GAEAlimited** [comment]
If you are working with Flutter, you can use the web3modal_flutter package to integrate Web3Modal into your Flutter app. 
Here's a simple example:

First, add the web3modal_flutter package to your pubspec.yaml file: 
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/69316708/df2bdc38-00e8-4393-9f2a-17e7bc8585f5)

Then, run flutter pub get to fetch the package.

Now you can use the following example to connect to a wallet using Web3Modal and send a transaction:
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/69316708/1b8713f8-6796-429d-a89d-be69027fdd65)

In this example:

We use the web3modal_flutter package to create a Web3Controller.
The _connectWallet function is used to connect to a wallet.
The _sendTransaction function demonstrates how to send a transaction. Adjust the recipient address, gas price, and value according to your requirements.
Remember to handle errors and loading states appropriately in a production environment. Also, make sure to use the latest versions of the packages by checking pub.dev for updates.

**Comment by 124maa** [comment]
احصل على Outlook for Android<https://aka.ms/AAb9ysg>

**Comment by 124maa** [comment]
***@***.***
